src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -216,50 +219,113 @@ public void checkServerTrusted(X509Certificate[] chain,
         return context;
     }
 
+    private static void completeResponse(Promise<Response> promise,
+                                         RequestOptions requestOptions,
+                                         IResponseCallback callback,
+                                         HttpResponse httpResponse,
+                                         HttpContext httpContext) {
+        try {
+            Map<String, String> headers = new HashMap<>();
+            for (Header h : httpResponse.getAllHeaders()) {
+                headers.put(h.getName().toLowerCase(), h.getValue());
+            }
+            String origContentEncoding = headers.get("content-encoding");
+            if (requestOptions.getDecompressBody()) {
+                new ResponseContentEncoding().process(httpResponse, httpContext);
+            }
+            Object body = null;
+            HttpEntity entity = httpResponse.getEntity();
+            if (entity != null) {
+                body = entity.getContent();
+            }
+            ContentType contentType = null;
+            if (headers.get("content-type") != null) {
+                contentType = ContentType.parse(headers.get("content-type"));
+            }
+            if (requestOptions.getAs() == ResponseBodyType.TEXT) {
+                body = coerceBodyType((InputStream) body, requestOptions.getAs(), contentType);
+            }
+            deliverResponse(requestOptions,
+                    new Response(requestOptions, origContentEncoding, body,
+                            headers, httpResponse.getStatusLine().getStatusCode(),
+                            contentType),
+                    callback, promise);
+        } catch (Exception e) {
+            deliverResponse(requestOptions, new Response(requestOptions, e), callback, promise);
+        }
+    }
+
+    private static void executeWithConsumer(final CloseableHttpAsyncClient client,
+                                            final FutureCallback<HttpResponse> futureCallback,
+                                            final HttpRequestBase request) {
+        /*
+         * Create an Apache AsyncResponseConsumer that will return the response to us as soon as it is available,
+         * then send the response body asynchronously
+         */
+        final StreamingAsyncResponseConsumer consumer = new StreamingAsyncResponseConsumer(new Deliverable<HttpResponse>() {
```

Comment:
this line is really long (I really hate having to scroll github's view, and also it's incredibly long when viewed in an editor), can it be made shorter easily?